### PR TITLE
Checker functions for orthogonality of coordinate system

### LINKED ISSUE #12852: Vector module: make sure that the base vectors remain orthogonal
Given a generic transformation equation, make sure that the base vectors remain orthogonal, if not, raise an exception.

### PR DESCRIPTION
**Pull request by szymag**
This PR introduce additional condition for transformation equation which must be fulfill. 
In `vector` module we are dealing with orthogonal curvilinear coordinate system only, so we need to check if transformation equations are correctly defined.

Closes #12852

### CODE REVIEW

**Comment by szymag** [comment]
ping @Upabjojr

**Comment by Upabjojr** [comment]
Need tests.

File: sympy/vector/coordsysrect.py [hunk 1]
```diff
@@ -213,6 +214,37 @@ def _connect_to_standard_cartesian(self,  curv_coord_type):
...
+
+        """
+
+        eq = self._transformation_equations()
+
+        v1 = Matrix([diff(eq[0], self.x), diff(eq[1], self.x), diff(eq[2], self.x)])
+        v2 = Matrix([diff(eq[0], self.y), diff(eq[1], self.y), diff(eq[2], self.y)])
+        v3 = Matrix([diff(eq[0], self.z), diff(eq[1], self.z), diff(eq[2], self.z)])
+
+        if any(simplify(i[0]+i[1]+i[2]) == 0 for i in (v1, v2, v3)):
```
Comment (Upabjojr) [inline sympy/vector/coordsysrect.py]: This condition is technically the condition of non-degenerancy, it is not related to the orthogonality.

Comment (szymag) [inline sympy/vector/coordsysrect.py]: Yes, but we need to check that. `Dot` product with degenerated vector gives `0` and we get information that system is orthogonal but it isn't true.
